components/root/root.jsx

```diff
@@ -319,6 +319,10 @@ export default class Root extends React.Component {
                         path={'/:team'}
                         component={NeedsTeam}
                     />
+                    <LoggedInRoute
+                        path={'/'}
+                        component={NeedsTeam}
```

Comment:
0/5 but by just looking at the path and component, it tells the reader differently :smile:.  Weird that the component's name needs a team but we're not passing a `:team` on path.

Reply:
You bring up a good point. I don't think the actual component here matters. From what I can tell, the reason that this works is because:
1. When you're logged in, the Root component sends you to `/teams/some-team`
2. When you're not logged in, the LoggedIn component (used by LoggedInRoute) sends you to `/login?redirect_to=%2F`. Notably, this also adds the `?redirect_to=%2F` to the end of the URL when not logged in which is a bit ugly.

I think we should try to make it more clear what we're trying to do with this code since we're sending a non-logged-in user to a section of the code that requires a session, so that causes problems like the websocket attempting to connect when someone goes to log in.

I don't know if showing the login page to logged in users is a new issue, but the root problem seems to me to be that we're even reaching this point in the code. `this.redirectIfNecessary` should be preventing a logged in user from even reaching the `/` route, and I think it's because `GlobalActions.redirectUserToDefaultTeam` is running asynchronously. If we wait for that to complete in `this.onConfigLoaded` before setting `this.state.configLoaded` to true, we shouldn't even hit this route.

Reply:
@hmhealey Those are some valid concerns. I will try and address them. The component as such does not make any difference as you said. Just that I was optimistically loading the javascript for team route as that is the most common use case(Complete assumption here based on personal experience). I see a quite a few 401's with this code as opposed to master when I logout so, that is definitely not desirable. 

My initial thought about fixing this is to load team route optimistically than taking users to login route and redirect to team route. Though you don't see the flash always, Behind the scenes you can see the route change happening in that order which is not desirable as well. 

my fix is too naive for this problem i like the idea of using the existing state for this as you suggested. I need some to think about the solution as there are blocking states in subsequent components as well. As this is not pressing issue i will think of a decent solution so as to not make more bugs around it.

Reply:
@saturninoabril That's odd too. I will rethink this and get back to you., Thanks